### PR TITLE
Migrate pkg/kubelet/server to structured logging

### PR DESCRIPTION
**Pull request by adisky**
Signed-off-by: Aditi Sharma <adi.sky17@gmail.com>

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Migrate pkg/kubelet/server to structured logging.

#### Which issue(s) this PR fixes:

part of #98976 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: <link> KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
```

### CODE REVIEW

File: pkg/kubelet/server/server.go [hunk 1]
```diff
@@ -175,7 +180,11 @@ func ListenAndServeKubeletReadOnlyServer(host HostInterface, resourceAnalyzer st
 		Handler:        &s,
 		MaxHeaderBytes: 1 << 20,
 	}
-	klog.Fatal(server.ListenAndServe())
+
+	if err := server.ListenAndServe(); err != nil {
+		klog.ErrorS(err, "Failed to listen amd serve")
```
Comment (navidshaikh) [inline pkg/kubelet/server/server.go:185]:
```suggestion
		klog.ErrorS(err, "Failed to listen and serve")
```

File: pkg/kubelet/server/server.go [hunk 2]
```diff
@@ -159,9 +160,13 @@ func ListenAndServeKubeletServer(
 		// Passing empty strings as the cert and key files means no
 		// cert/keys are specified and GetCertificate in the TLSConfig
 		// should be called instead.
-		klog.Fatal(s.ListenAndServeTLS(tlsOptions.CertFile, tlsOptions.KeyFile))
-	} else {
-		klog.Fatal(s.ListenAndServe())
+		if err := s.ListenAndServeTLS(tlsOptions.CertFile, tlsOptions.KeyFile); err != nil {
+			klog.ErrorS(err, "Failed to listen and serve.")
```
Comment (navidshaikh) [inline pkg/kubelet/server/server.go:164]:
```suggestion
			klog.ErrorS(err, "Failed to listen and serve")
```

File: pkg/kubelet/server/server.go [hunk 3]
```diff
@@ -185,9 +194,12 @@ func ListenAndServePodResources(socket string, podsProvider podresources.PodsPro
 	podresourcesapi.RegisterPodResourcesListerServer(server, podresources.NewV1PodResourcesServer(podsProvider, devicesProvider, cpusProvider))
 	l, err := util.CreateListener(socket)
 	if err != nil {
-		klog.Fatalf("Failed to create listener for podResources endpoint: %v", err)
+		klog.ErrorS(err, "Failed to create listener for podResources endpoint")
```
Comment (serathius) [inline pkg/kubelet/server/server.go:197]: When replacing `Fatalf`  we should add `os.Exit(1)` to maintain exit. Same as you did above when replacing `Fatal`

**Comment by adisky** [comment]
/cc @serathius @ehashman 
Can you review? Not sure how to migrate` Fatal` calls to `ErrorS`

File: pkg/kubelet/server/server.go [hunk 4]
```diff
@@ -185,9 +194,12 @@ func ListenAndServePodResources(socket string, podsProvider podresources.PodsPro
...
+	}
+
+	if err := server.Serve(l); err != nil {
+		klog.ErrorS(err, "Failed to Serve")
```
Comment (serathius) [inline pkg/kubelet/server/server.go:201]: When replacing `Fatalf`  we should add `os.Exit(1)` to maintain exit. Same as you did above when replacing `Fatal`

Comment (navidshaikh) [inline pkg/kubelet/server/server.go:201]:
```suggestion
		klog.ErrorS(err, "Failed to serve")
```

**Review by navidshaikh — CHANGES_REQUESTED** [review]

**Comment by adisky** [comment]
Thanks for the review @serathius @navidshaikh 
Addressed review comments PTAL

**Review by navidshaikh — APPROVED** [review]
lgtm

**Pull request merged**